packages/ember-glimmer/lib/utils/iterable.js

```diff
@@ -175,10 +175,18 @@ class EachInIterable {
     }
 
     let typeofIterable = typeof iterable;
-
     if (iterable && (typeofIterable === 'object' || typeofIterable === 'function')) {
-      let keys = Object.keys(iterable);
-      let values = keys.map(key => iterable[key]);
+      let keys = [];
+      let values = [];
+      if (HAS_NATIVE_MAP && iterable instanceof Map) {
```

Comment:
My thought was that we could simplify this a bit. Basically, use native symbol when possible then fallback to `.forEach` if present, else do `Object.keys`. And avoid directly doing instanceof checks here.

```js
const HAS_NATIVE_SYMBOL = (function() {
  let hasSymbol = typeof Symbol === 'function';
  if (hasSymbol === false) { return false; }

  let instance = Symbol('test-symbol');
  // use `Object`'s `.toString` directly to prevent us from detecting
  // polyfills as native
  return Object.prototype.toString.call(instance) === '[object Symbol]';
})();

// ...snip...

if (iterable && (typeofIterable === 'object' || typeofIterable === 'function')) {
  let keys;
  let values;

  if (HAS_NATIVE_SYMBOL && typeof iterable[Symbol.iterator]) {
    keys = [];
    values = [];
    let iterator = iterable[Symbol.iterator]();
    
    while(true) {
      let result = iterator.next();
      if (result.done) break;
      let [key, value] = result.value;
      keys.push(key);
      values.push(value);
    }
  } else if (typeof iterable.forEach === 'function') {
    keys = [];
    values = [];
    iterable.forEach((value, key) => {
      keys.push(key);
      values.push(value);
    });
  } else {
    keys = Object.keys(iterable);
    values = keys.map(key => iterable[key]);  
  }
}
// ...snip...
```